inc/Calculator.h

```diff
@@ -0,0 +1,394 @@
+#ifndef CALCUALTOR
+#define CALCULATOR
+#define RAYGUI_IMPLEMENTATION
+
+#include <string>
+#include "Widget.h"
+
+#include <raylib.h>
+#include <iostream>
+#include <ctime>
+#include "Constants.h"
+#include <stack>
+#include <list>
+#include <cmath>
+#include <iomanip>
+using namespace std;
+
+class Calculator : public Widget {
+std::string text;
+Vector2 mousePoint;
+Texture2D plusButton;
+bool Button1Pressed;
+int offset;
+int lastOperatorIndex;
+
+stack<char> operators;
+stack<float> operands;
+public:
+    Calculator(uint32_t x, uint32_t y, uint32_t w, uint32_t h, std::string n) :
+        Widget (x, y, w, h, n) {
+        // Calls widget's constructor
+        //render = false;
+        text = "";
+        mousePoint = {0.0f, 0.0f};
+        Image plusIcon = LoadImage("../resources/images/plus_operator.png");
+        ImageResize(&plusIcon, 50, 50);
+        plusButton = LoadTextureFromImage(plusIcon);
+        UnloadImage(plusIcon);
+        Button1Pressed = false;
+        offset = 25;
+        
+    }
+    void spawnButton(int x, int y, int w,int h, std::string buttonText)
+    {
+        DrawRectangleLines(x,y,w,h,SKYBLUE);
+        int buttonOffset = (50 - MeasureText(buttonText.c_str(),30))/2;
+        DrawText(buttonText.c_str(), buttonOffset+x, 10 + y, 30, SKYBLUE);
+    }
+    bool buttonInteract(int x, int y, int w,int h, std::string buttonText){
+        if (CheckCollisionPointRec(mousePoint, (Rectangle){x,y,w,h})){
+        DrawRectangle(x,y,w,h,SKYBLUE);
+        int buttonOffset = (50 - MeasureText(buttonText.c_str(),30))/2;
+        DrawText(buttonText.c_str(), buttonOffset+x, 10 + y, 30, BLACK);
+
+        if (IsMouseButtonDown(MOUSE_BUTTON_LEFT)) return true;
+            else return false;
+        }
+        
+
+    }
+
+    /*int prec(char ch) {
```

Comment:
Comment these functions and previous functions to ensure that future you knows what you've programmed as well as other developers